src/setting_sbp_cb.c

```diff
@@ -220,15 +213,16 @@ static void setting_write_resp_callback(uint16_t sender_id,
     return;
   }
 
-  if (setting_update_watch_only(ctx, write_response->setting, len - sizeof(write_response->status))
-      != 0) {
-    log_warn("error in settings read response message");
-  }
+  /* Update watchers, no need for filter as wathers shall not be read-only */
+  setting_value_update(ctx,
+                       write_response->setting,
+                       len - sizeof(write_response->status),
+                       UPDATE_FILTER_NONE);
```

Comment:
this refactor doesn't quite retain the original intent of `setting_update_watch_only` which was to write the value to a setting if and only if that setting is watch only. now that the code is centralized it is much easier to reason about this so I'm less worried. maybe we can just add a `UPDATE_FILTER_NORMAL` which I suppose would return on the following condition: `!setting_data->readonly && !setting_data->watchonly`. open to other ideas too

Reply:
Seems fine for now. If there's anymore properties in the future than read/watchonly, then reconsider.